ctapipe/io/hessioeventsource.py

```diff
@@ -161,10 +161,27 @@ def _generator(self):
                                                        get_azimuth_raw(tel_id))
                     data.mc.tel[tel_id].altitude_raw = (file.
                                                         get_altitude_raw(tel_id))
-                    data.mc.tel[tel_id].azimuth_cor = (file.
-                                                       get_azimuth_cor(tel_id))
-                    data.mc.tel[tel_id].altitude_cor = (file.
-                                                        get_altitude_cor(tel_id))
+                    azimuth_cor = file.get_azimuth_cor(tel_id)
+                    altitude_cor = file.get_altitude_cor(tel_id)
+
+                    # hessioeventsource pass 0 if there is no altitude/azimuth correction
+                    if azimuth_cor == 0 and data.mc.tel[tel_id].azimuth_raw != 0:
+                        data.mc.tel[tel_id].azimuth_cor = np.nan
+                        data.pointing[tel_id].azimuth = \
```

Comment:
Please use () rather than `\` for continuation.  Actually, perhaps just run "black" on this before committing - it tends to fix all style problems.  I think we should make that a policy (or even commit hook) anyhow, to avoid style changes.  The default settings.

Reply:
In the simteleventsour whe have `pointing = data.pointing[tel_id]` and `mc = `data.mc.tel[tel_id]` in the telescope loop so the lines get much shorter and more readable. That's something black won't do ;-)

Reply:
yes, also simplifying the long lines is a better way to go! .  Even just defining `tel = data.mc.tel[tel_id]`, and `pointing = data.pointing[tel_id]`, and then using those  references would make this easier to follow.

Reply:
Ok I cleaned the whole loop following @MaxNoe suggestion. Looks more clear.